test/graphkb.test.js

```diff
@@ -0,0 +1,68 @@
+const { simplifyRecordsLinks } = require('../src/graphkb');
+
+describe('simplifyRecordsLinks', () => {
+    const listOfContentToReturnAsIs = [
+        1.0,
+        null,
+    ];
+    const listOfContentToReturnUnnested = [
+        {},
+        { a: 1, b: 2 },
+        { '@rid': 123, a: 1 },
+        {
+            a: [
+                { '@rid': 123, aa: 1 },
+                { ab: 2 },
+            ],
+        },
+        {
+            a: { '@rid': 123, aa: 1 },
+            b: 2,
+        },
+        {
+            a: {
+                '@rid': 123,
+                a: { '@rid': 123, aa: 1 },
+            },
+        },
+    ];
+    const ListOfReturnedUnnestedContent = [
+        {},
+        { a: 1, b: 2 },
+        { '@rid': 123, a: 1 },
+        {
+            a: [
+                '123',
+                { ab: 2 },
+            ],
+        },
+        { a: '123', b: 2 },
+        { a: '123' },
+    ];
+
+    test.each(listOfContentToReturnAsIs)('do nothing when content not a list/k-v pair', content => {
+        const output = simplifyRecordsLinks(content);
+        expect(output).toBe(content);
+    });
+
+    const cases = [];
+    listOfContentToReturnUnnested.forEach((content, index) => {
+        cases.push([content, ListOfReturnedUnnestedContent[index]]);
+    });
+
+    test.each(cases)('simplifies nested obj. to RID', (content, expectedOutput) => {
```

Comment:
you can simplify this by just passing an array of arrays to test.each. I would also avoid putting them in variables at all since you are not re-using them anywhere

see example here: https://github.com/bcgsc/pori_graphkb_parser/blob/develop/test/index.test.ts#L93

so here that would look something like this for the first case

```js
test.each([
  1.0,
  null,
  {},
  {a: 1, b: 1},
])('does not change', (inputValue) => {
    const output = simplifyRecordsLinks(inputValue);
    expect(output).toEquall(inputValue);
});
```

Note I would add some more examples here like a string input or an object that doesn't change which is why i have used the toEqual functon. https://jestjs.io/docs/expect#toequalvalue

for the second one you do the same thing but just expect multiple args and pass an array of arrays, I've used ... as a placeholder to avoid copyig the other examples

```js
test.each([
  [{ a: [  { '@rid': 123, aa: 1 }, { ab: 2 }, ], },, { a: [ 123, { ab: 2 }, ], },],
  [...]
])('does not change', (inputValue, expectedValue) => {
    const output = simplifyRecordsLinks(inputValue);
    expect(output).toEquall(exoectedValue);
});
```